### ISSUE TITLE
Self-referential Autoload crashes on project load

### ISSUE THREAD

**Issue by gokiburikin** [godotengine/godot#55843]
### Godot version

v4.0.dev.20211210.official [21b81998b]

### System information

Windows 10

### Issue description

Self referential autoloads crash in the latest pre-alpha release. This does not happen with previous pre-alpha builds I've tested.

### Steps to reproduce

Create a new project, create a new script with the following code and add it as an autoload named Foo:

```gdscript
extends Node
var foo = Foo
```

### Minimal reproduction project

[self-ref-autoload-mrp.zip](https://github.com/godotengine/godot/files/7697920/self-ref-autoload-mrp.zip)

**Comment by KoBeWi** [comment]
Can't reproduce the crash on newest master. The project loads normally, but has an error in code (it can't find autoload, because it couldn't add autoload, because there is error that it can't add autoload). After fixing this problem there is still no crash.

**Comment by gokiburikin** [comment]
Some additional information: The latest nightly also handles this case with no issue including the projects that originally crashed on load which led me to create the MRP. The version I reported the crash on I tested specifically because it was [posted to the subreddit](https://www.reddit.com/r/godot/comments/rdhj2n/godot_40_prealpha_snapshot_from_20211210/), but it looks like there just happened to be some autoload issue in that specific build.

For completeness' sake I also tested the prior pre-alpha release posted to the subreddit and it also does not crash.

Seeing as it doesn't happen in current builds or prior builds this is likely no longer an issue.
